lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
+        Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(cookbook_path), "*")).each do |dir|
```

Comment:
and as a rule anywhere you see `Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(...)))` that signature implies that the inside of the loop takes forward slashes just fine (because of the File.join outside of the escape_glob) so that should always be converged to escape_glob_dir and everything inside of the Dir.glob should be using forward slashes.